src/audio/base_fw_intel.c

```diff
@@ -298,6 +356,27 @@ static int basefw_set_fw_config(bool first_block,
 	return 0;
 }
 
+static int basefw_dma_control(bool first_block, bool last_block,
+			      uint32_t data_size, const char *data)
+{
+	struct ipc4_dma_control *dma_control;
+	int ret;
+
+	if (!(first_block && last_block))
+		return IPC4_INVALID_REQUEST;
+
+	dma_control = (struct ipc4_dma_control *)data;
+
+	if (data_size < offsetof(struct ipc4_dma_control, config_data[dma_control->config_length]))
+		return IPC4_INVALID_CONFIG_DATA_STRUCT;
+
+	ret = dai_control(dma_control->node_id, data);
+	if (ret)
+		return IPC4_INVALID_REQUEST;
+
+	return 0;
```

Comment:
there's an IPC4 return code for "success"

Reply:
Used explicit `IPC4_SUCCESS` code instead of 0.